internal/pkg/gateway/api.go

```diff
@@ -321,7 +337,7 @@ func (gs *Server) Endorse(ctx context.Context, request *gp.EndorseRequest) (*gp.
 		return nil, newRpcError(codes.Aborted, "failed to collect enough transaction endorsements, see attached details for more info", errorDetails...)
 	}
 
-	env, err := protoutil.CreateTx(proposal, endorsements...)
```

Comment:
But previous this was the only place in the code that was using protoutil.CreateTx, so shouldn't we also erase CreateTx altogether?

Reply:
Yep, although given the feedback, I'll rework it so that it shares as much common code as possible.  I am trying to avoid the duplicate unmarshalling that's currently happening in the gateway,  but I can rework the 2 wrapper functions in protoutil to avoid that.